heed/src/env.rs

```diff
@@ -260,6 +261,57 @@ pub enum CompactionOption {
 }
 
 impl Env {
+    /// Return the size used by all the databases in the environment.
+    pub fn size(&self) -> Result<usize> {
+        let mut size = 0;
+
+        let mut stat = std::mem::MaybeUninit::uninit();
+        unsafe { mdb_result(ffi::mdb_env_stat(self.0.env, stat.as_mut_ptr())) }?;
+        let stat = unsafe { stat.assume_init() };
+
+        size += (stat.ms_leaf_pages + stat.ms_branch_pages + stat.ms_overflow_pages)
+            * stat.ms_psize as usize;
```

Comment:
Could we create a _local_ closure to do this computation? instead of duplicating the code two times?